android/src/toga_android/widgets/label.py

```diff
@@ -26,12 +26,6 @@ def set_color(self, value):
             self.native.setTextColor(native_color(value))
 
     def set_textview_alignment(self, value, vertical_gravity):
-        # Refuse to set alignment if widget has no container.
-        # On Android, calling setGravity() when the widget has no LayoutParams
-        # results in a NullPointerException.
-        if not self.native.getLayoutParams():
-            return
```

Comment:
This check was preventing the widget's initial alignment from taking effect. But the testbed and the example app both work fine without it, on both API level 24 (our current minimum) and 33 (the current stable version).

So I guess that either this only affected older versions of Android, or the relative timing of container addition and style application has changed in Toga itself.

Reply:
I'm fairly certain this was a quirk of an early iteration of Toga's layout on Android. There's a bunch of these warts on every widget (usually on the rehint method). I would definitely like to clean these up if we can.